interfaces/inc/c_pal/malloc_multi_flex.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+#ifndef MALLOC_MULTI_FLEX_H
+#define MALLOC_MULTI_FLEX_H
+
+#ifdef __cplusplus
+#include <cstddef>
+#include <cstdint>
+#else
+#include <stddef.h>
+#include <stdint.h>
+#endif
+
+#include "umock_c/umock_c_prod.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+    #define ARG_OVERFLOW_CHECK(arg) \
+        if ((MU_C2(arg, _size) != 0 && SIZE_MAX / MU_C2(arg, _size) < MU_C2(arg, _count)) || (SIZE_MAX - size_tracker < MU_C2(arg, _count) * MU_C2(arg, _size)))\
+        {\
+            return NULL;\
+        }\
+        size_tracker += MU_C2(arg, _count) * MU_C2(arg, _size);
+
+    #define ARGS_OVERFLOW_CHECK(...) \
+        size_t size_tracker = parent_struct_size;\
+        MU_FOR_EACH_1(ARG_OVERFLOW_CHECK, __VA_ARGS__)
+
+    #define ARG_LIST_VALUE_ARG(arg) \
+        , uint32_t MU_C2(arg, _count), uint32_t MU_C2(arg, _size)
+
+    #define ARG_LIST_VALUES(...) \
+        MU_FOR_EACH_1(ARG_LIST_VALUE_ARG, __VA_ARGS__)
+
+    #define ASSIGN_INTERNAL_STRUCT_PTR(arg) \
+        parent_struct_pointer->arg = pointer_iterator;\
+        pointer_iterator = (char*)pointer_iterator + MU_C2(arg, _count) * MU_C2(arg, _size);;
```

Comment:
>(char*) [](http://example.com/codeflow?start=27&length=7)

Nit: char* makes one think we're dealing with a string, I'd use uint8_t*

Reply:
uintptr_t has been used in the latest iteration